backend/tests/integration/pidstore/minters/test_minters_cnum.py

```diff
@@ -189,3 +189,43 @@ def test_minter_mints_cnum_on_update_when_cnum_is_missing_in_db(
     expected_cnum = "C05-01-01"
     assert 1 == len(record_cnums)
     assert record_cnums[0].pid_value == expected_cnum
+
+
+def test_generate_cnum_when_holes_in_cnums_sequence(base_app, db, create_record):
```

Comment:
maybe also add a test where you mint `C20-01-01.1` because only `C20-01-01` exists as it's a different case?

Reply:
I think that this is actually tested already on the old tests like here: test_minter_mint_cnum_more_than_once